matrix_content_scanner/scanner/scanner.py

```diff
@@ -59,14 +93,140 @@ async def scan_file(
         also cache the result.
 
         If the file already has an entry in the result cache, return this value without
-        downloading the file again.
+        downloading the file again (unless we purposefully did not cache the file's
+        content to save up on memory).
+
+        Args:
+            media_path: The `server_name/media_id` path for the media.
+            metadata: The metadata attached to the file (e.g. decryption key), or None if
+                the file isn't encrypted.
+            thumbnail_params: If present, then we want to request and scan a thumbnail
+                generated with the provided parameters instead of the full media.
+
+        Returns:
+            A description of the media.
+
+        Raises:
+            ContentScannerRestError if the file could not be downloaded.
+            FileDirtyError if the result of the scan said that the file is dirty, or if
+                the media path is malformed.
+        """
+        # Compute the cache key for the media.
+        cache_key = self._get_cache_key_for_file(media_path, metadata, thumbnail_params)
+
+        # The content of the file, if we have already scanned it. We might re-download
+        # a file that was too large to cache, and then realise we need to scan it again
+        # (because e.g. its hash has changed), in which case we set this to the file's
+        # MediaDescription and give it to _download_and_scan to avoid it being scanned
+        # again.
+        media: Optional[MediaDescription] = None
+
+        # Return the cached result if there's one.
+        cache_entry = self._result_cache.get(cache_key)
+        if cache_entry is not None:
+            logger.info("Found a cached result %s", cache_entry.result)
+
+            if cache_entry.result is False:
```

Comment:
I find it clearer to have `is [bool value]` conditions to make it clearer that we're looking at the value of `result` rather than e.g. the existence of a value etc

Reply:
I don't really agree that your way is clearer, but I don't feel strongly.